src/core/tsi/s2a/record_protocol/s2a_crypter.h

```diff
@@ -33,6 +33,21 @@
  *  functionality. The interface is thread-compatible. **/
 typedef struct s2a_crypter s2a_crypter;
```

Comment:
I did this so that the function signatures don't all look like `struct s2a_crypter* crypter`. This seems to be the convention that is followed across the gRPC C-core (for example, see [alts_handshaker_client.h](https://github.com/grpc/grpc/blob/master/src/core/tsi/alts/handshaker/alts_handshaker_client.h)).

Reply:
This is only required for C, not for C++. You might still need it where you expose a C interface, but not for internal definitions.
Can discuss this offline and handle in a later PR if required.